cpp/include/cudf/strings/detail/copy_if_else.cuh

```diff
@@ -33,29 +33,28 @@ namespace detail {
  * strings from the lhs iterator or the rhs iterator.
  *
  * ```
- * output[i] = filter_fn(i) ? lhs(i).first : rhs(i).first
+ * output[i] = filter_fn(i) ? lhs(i) : rhs(i)
  * ```
  *
- * @tparam StringPairIterLeft Pair iterator returning thrust::pair<string_view,bool> where the
- *         bool parameter specifies if the string_view is valid (true) or not (false).
- * @tparam StringPairIterRight Pair iterator returning thrust::pair<string_view,bool> where the
- *         bool parameter specifies if the string_view is valid (true) or not (false).
+ * @tparam StringIterLeft An optional-iterator returning thrust::optional<string_view> where the
+ *         the result is a string_view only if the element is valid.
```

Comment:
This would be a bit more canonical way to describe it. 
```suggestion
 * @tparam StringIterLeft A random access iterator whose value_type is `thrust::optional<string_view>` where the
 *         `optional` has a value iff the element is valid. 
```